injectproxy/routes.go

```diff
@@ -204,17 +206,30 @@ func (hff HTTPFormEnforcer) ExtractLabel(next http.HandlerFunc) http.Handler {
 			}
 		}
 
-		next.ServeHTTP(w, r.WithContext(WithLabelValue(r.Context(), labelValue)))
+		next.ServeHTTP(w, r.WithContext(WithLabelValues(r.Context(), labelValues)))
 	})
 }
 
-func (hff HTTPFormEnforcer) getLabelValue(r *http.Request) (string, error) {
-	formValue := r.FormValue(hff.ParameterName)
-	if formValue == "" {
-		return "", fmt.Errorf("the %q query parameter must be provided", hff.ParameterName)
+func (hff HTTPFormEnforcer) getLabelValues(r *http.Request) ([]string, error) {
+	err := r.ParseForm()
+	if err != nil {
+		return nil, fmt.Errorf("the query parameter can not be parsed: %w", err)
+	}
+
+	formValues := r.Form[hff.ParameterName]
+
+	if len(formValues) == 0 {
+		return nil, fmt.Errorf("the %q query parameter must be provided", hff.ParameterName)
+	}
+
+	// Check if every value is non-empty
+	for i := range formValues {
+		if formValues[i] == "" {
+			return nil, fmt.Errorf("the %q query parameter must be provided", hff.ParameterName)
```

Comment:
I wonder if we shouldn't be liberal and skip empty values instead.

Reply:
Agreed. I have similar feel that we are too aggressive to report an error here. I have also changed the behavior for static label value and for header. Please have a look as well.